### PR TITLE
Update for deps

### PR DESCRIPTION
**Pull request by Nosferican**
Closes #70 #71 #72 #73 #74 #75 #76

Basically, a PR to update a few things which have gathered some dust.

- Updates the StatsBase (StatsAPI) including `responsename`, updated `dof` definition, and docs (module is now StatsAPI)
- Updates the compat for deps
- Updates the codemeta to CFF (GitHub lead w/CFF as the preferred standard)
- Updated the CI (actions to newer releases)

I tried updating the `show(io, mime, obj)` instead of using the 2-args method but didn't work well so that is out of scope for this one.

### CODE REVIEW

File: Project.toml [hunk 1]
```diff
@@ -3,41 +3,44 @@ uuid = "4d6a76a9-bfbc-5492-8924-cf6ed7875f06"
...
 FillArrays = "1a297f60-69ca-5386-bcde-b61e274b549b"
 ForwardDiff = "f6369f11-7733-5829-9624-2563aa707210"
 JLLWrappers = "692b3bcd-3c85-4b1f-b108-f13ce0eb3210"
 LinearAlgebra = "37e2e46d-f89d-539d-b4ee-838fcccc9c8e"
+MLJBase = "a7f614a8-145f-11e9-1d2a-a57a1082229d"
 MLJModelInterface = "e80e1ace-859a-464e-9ed9-23947d8ae3ea"
 Optim = "429524aa-4258-5aef-a3af-852621145aeb"
 Parameters = "d96e819e-fc66-5662-9728-84c9c7592b0a"
 Printf = "de0858da-6303-5e67-8744-51eddeeeb8d7"
+RDatasets = "ce6b1742-4840-55fa-b093-852dadbb1d8b"
```
Comment (giordano) [inline Project.toml:22]: I'm confused, why is this adding dependencies which are unused?

File: Project.toml [hunk 2]
```diff
@@ -3,41 +3,44 @@ uuid = "4d6a76a9-bfbc-5492-8924-cf6ed7875f06"
...
 [compat]
-CategoricalArrays = "0.8, 0.9"
-Distributions = "0.23, 0.24"
+CategoricalArrays = "0.8, 0.9, 0.10"
+Distributions = "0.23, 0.24, 0.25"
 FillArrays = "0.9, 0.10, 0.11"
 ForwardDiff = "0.10"
-MLJBase = "0.15, 0.16"
-MLJModelInterface = "0.3"
+JLLWrappers = "1"
```
Comment (giordano) [inline Project.toml:34]: This must be the first time I see a compat with `JLLWrappers` outside of a JLL package.

Comment (Nosferican) [reply Project.toml:34]: CompatHelper gets mad at me if I don't I guess. Guess it should work the same as any other package but probably not vital.

File: Project.toml (hunk 1 above)
Comment (Nosferican) [reply Project.toml:22]: Good catch, I had updated my environment while updating the tests. Going to try to get the docs working properly again and will clean that up before merging.

**Pull request merged**